cmd/minimega/disk.go

```diff
@@ -343,6 +355,37 @@ func parseInjectPairs(files []string) (map[string]string, error) {
 	return pairs, nil
 }
 
+// parseFiles parses the files argument passed into the 'disk inject ...'
+// command. if options are used, the files argument gets turned into a
+// minicli.Command StringArg, otherwise it gets turned into a ListArg.
+// this function returns either a paths string slice if 'delete' is part
+// of the original command, or a pairs string map.
+func parseFiles(files interface{}) (map[string]string, []string, error) {
+	var pairs map[string]string
+	var paths []string
+	var err error
+	switch v := files.(type) {
+	case []string:
+		if !(sliceContainsCharacters(v, ",:")) {
+			paths = []string{v[0]}
+		} else if sliceContainsString(v, ",") {
+			paths = strings.Split(v[0], ",")
+		} else {
+			pairs, err = parseInjectPairs(v)
+		}
+	case string:
+		if strings.Contains(v, ",") {
```

Comment:
Can the single case also be added here. Otherwise it doesn' t work with options specified

```
minimega[minimega]$ disk inject /scratch/images/test.qcow2 options "-o offset=0" delete files /bin/foo
Error (ccc19.40g): malformed command; expected src:dst pairs
```

Reply:
Fixed